src/generate/api.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { model } from "amf-client-js";
+import _ from "lodash";
+
+import {
+  resolveApiModel,
+  parseRamlFile,
+  getAllDataTypes
+} from "../common/parser";
+
+/**
+ * An API represented as an AMF model. Includes the extracted data types
+ * defined in the spec. Common transformations of the name are cached for
+ * reference in templates and for use in assembling file paths.
+ */
+export class Api {
```

Comment:
MIght want a field pointing back at the group.  That way an api knows what group it is a part of.

Reply:
I like this on the surface, but then I started to think about the code to manage keeping it in sync and I don't love that part (we'd have to wrap up the array in ApiGroup). How much value do you think it is?

Reply:
Well let me rephrase my concern and you let me know how to solve it.

First...Im not too concerned about drift here since we aren't rewriting data much...once it is read it is fairly immutable for that generation run.

Secondly, if you look what I was doing around inline documentation and example code we need to have some bread crumb of where it came from.  This is the primary aspect I am trying to solve.

Reply:
Api now has a string field for 'group' that is set when init'ing group